modules/folder/variables-iam.tf

```diff
@@ -51,6 +51,51 @@ variable "iam_bindings_additive" {
   default  = {}
 }
 
+variable "iam_by_principals_additive" {
+  description = "Additive IAM binding in {PRINCIPAL => [ROLES]} format. Principals need to be statically defined to avoid cycle errors."
+  type        = map(list(string))
+  default     = {}
+  nullable    = false
+
+  validation {
+    error_message = format(
+      "Cannot refer to the same principal/role pair from iam_by_principals_additive and iam_bindings_additive. The following pairs are present on both sides:\n%s",
```

Comment:
I would skip the format and just spit out a generic message: it should be pretty simple for the user to figure out the dupes. We never use context-aware validation messages, IMHO they add unnecessary complexity to what is essentially just a failsafe.

Reply:
Actually, I'll just remove the validation